### PR TITLE
feat: Start/stop commands

### PR DESCRIPTION
**Pull request by jonas-martinez**
Closes https://github.com/lenra-io/lenra_cli/issues/3

## Checklist
- [x] I didn't over-scope my PR
- [x] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I did not include breaking changes
- [ ] I included unit tests that cover my changes
- [x] I added/updated the documentation about my changes
- [x] I made my own code-review before requesting one

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name = "lenra_cli"
 version = "0.0.0"
 edition = "2021"
 license = "MIT"
-description = "THe Lenra's command line interface"
+description = "The Lenra's command line interface"
```
Comment (jonas-martinez) [inline Cargo.toml:6]:
```suggestion
description = "The Lenra command line interface"
```

File: README.md [hunk 2]
```diff
@@ -36,7 +36,11 @@ The Lenra's command line interface.
 
 ### Prerequisites
 
-Install the Lenra's cli using one of the next possibilities.
+To build and run the Lenra elements that handle your app, the Lenra CLI needs [Docker](https://docs.docker.com/engine/install/) and [Docker Compose](https://docs.docker.com/compose/install/).
+
+You also can install the [Docker buildx command](https://docs.docker.com/build/buildx/install/) to use the [Buildkit optimization given by Dofigen](https://github.com/lenra-io/dofigen).
```
Comment (jonas-martinez) [inline README.md:41]:
```suggestion
You can also install the [Docker buildx command](https://docs.docker.com/build/buildx/install/) to use the [Buildkit optimization given by Dofigen](https://github.com/lenra-io/dofigen).
```

File: README.md [hunk 3]
```diff
@@ -69,7 +65,7 @@ Use the help options to understand how to use it:
 ```bash
 $ lenra --help
 lenra_cli 0.0.0
-A Dockerfile generator using a simplified description in YAML or JSON format command line tool
+The Lenra's command line interface
```
Comment (jonas-martinez) [inline README.md:68]:
```suggestion
The Lenra command line interface
```

File: README.md [hunk 4]
```diff
@@ -136,6 +135,40 @@ OPTIONS:
...
+```
+
+#### stop
+
+This subcommand stops the Lenra app of the current directory and removes the Docker Compose elements.
+
+```bash
+$ lenra stop --help
+lenra-stop 
+Stop your app previously started with the start command
```
Comment (jonas-martinez) [inline README.md:162]:
```suggestion
Stops your app previously started with the start command
```

File: src/cli.rs [hunk 5]
```diff
@@ -23,6 +23,8 @@ pub enum Command {
     Build(Build),
     /// Start your app previously built with the build command
     Start(Start),
+    /// Stop your app previously started with the start command
```
Comment (jonas-martinez) [inline src/cli.rs:26]:
```suggestion
    /// Stops your app previously started with the start command
```

File: README.md [hunk 6]
```diff
@@ -83,6 +79,7 @@ SUBCOMMANDS:
     help     Print this message or the help of the given subcommand(s)
     new      Create a new Lenra app project
     start    Start your app previously built with the build command
+    stop     Stop your app previously started with the start command
```
Comment (jonas-martinez) [inline README.md:82]:
```suggestion
    stop     Stops your app previously started with the start command
```

File: README.md [hunk 7]
```diff
@@ -91,6 +88,8 @@ This tools contains many subcommands to help you doing what you need.
 
 - [new](#new): creates a new Lenra app project
 - [build](#build): builds the Lenra app of the current directory
+- [start](#start): start your app previously built with the build command
```
Comment (jonas-martinez) [inline README.md:91]:
```suggestion
- [start](#start): starts your app previously built with the build command
```

File: src/stop.rs [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    /// Starts the docker-compose
+    fn stop_docker_compose(&self) {
+        let dockercompose_path: PathBuf = DOCKERCOMPOSE_DEFAULT_PATH.iter().collect();
+        if !dockercompose_path.exists() {
+            
+        }
+
+        let mut command = Command::new("docker");
+
+        // TODO: display std out & err
```
Comment (jonas-martinez) [inline src/stop.rs:26]: Did you leave this TODO for future work or do you have to do it in this PR ?

File: src/docker_compose.rs [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+                        depends_on: Some(DependsOnOptions::Conditional(
+                            [(
+                                DEVTOOL_SERVICE_NAME.into(),
+                                DependsCondition {
+                                    condition: "service_healthy".into(),
+                                },
+                            )]
+                            .into(),
+                        )),
+                        // TODO: Add resources management  when managed by the docker-compose-types lib
```
[the remark this answers is not in the document]
Comment (taorepoara) [reply src/docker_compose.rs:86]: It's for futur work since the docker compose types lib does not manage them yet.

File: src/cli.rs (hunk 5 above)
Comment (taorepoara) [reply src/cli.rs:26]: I'm not sure of this one. It's like the other commands.

The CLI are mostly explaining what "you" do using the command and what the command does.

**Review by taorepoara — APPROVED** [review]

**Pull request merged**